test/Serialization/comments-hidden.swift

```diff
@@ -0,0 +1,59 @@
+// Test the case when we compile normally
+//
+// RUN: rm -rf %t
+// RUN: mkdir %t
+// RUN: %target-swift-frontend -module-name comments -emit-module -emit-module-path %t/comments.swiftmodule -emit-module-doc -emit-module-doc-path %t/comments.swiftdoc %s
+// RUN: %target-swift-ide-test -print-module-comments -module-to-print=comments -source-filename %s -I %t | FileCheck %s -check-prefix=NORMAL
+
+// Test the case when we compile with -enable-testing
+//
+// RUN: rm -rf %t
+// RUN: mkdir %t
+// RUN: %target-swift-frontend -enable-testing -module-name comments -emit-module -emit-module-path %t/comments.swiftmodule -emit-module-doc -emit-module-doc-path %t/comments.swiftdoc %s
+// RUN: %target-swift-ide-test -print-module-comments -module-to-print=comments -source-filename %s -I %t | FileCheck %s -check-prefix=TESTING
+
+/// PublicClass Documentation
+public class PublicClass {
+  /// Public Function Documentation
+  public func f_public() { }
+  /// Internal Function Documentation
+  internal func f_internal() { }
+  /// Private Function Documentation
+  private func f_private() { }
+}
+
+/// InternalClass Documentation
+internal class InternalClass {
+  /// Internal Function Documentation
+  internal func f_internal() { }
+  /// Private Function Documentation
+  private func f_private() { }
+}
+
+/// PrivateClass Documentation
+private class PrivateClass {
+  /// Private Function Documentation
+  private func f_private() { }
+}
+
+// NORMAL-NOT: PrivateClass Documentation
```

Comment:
So, I took a look at this idea tonight.  What this PR does is take something like this:

```bash
$ swift-ide-test -target x86_64-apple-macosx10.9   -print-module-comments --source-filename foo.swift
Class/PrivateClass RawComment=// My great comment BriefComment=My Great comment DocCommentAsXML=...
```

and turns it into

```
Class/PrivateClass RawComment=none BriefComment=none DocCommentAsXML=none
```

In other words, **the string `Private` still appears**, it's just that the *documentation comment* for that symbol is removed.

Is eliminating the class *name* itself from the `-print-module-comments` output a good thing to do, if so, can we break that into a separate PR, and still look for `//NORMAL-NOT: PrivateClass Documentation` etc. in this test?